src/fosslight_util/download.py

```
@@ -1013,6 +1013,53 @@ def extract_rpm_payload(source_file: str, dest_path: str) -> bool:
+        for f in files:
+            fpath = os.path.join(root, f)
+            try:
+                os.chmod(fpath, os.stat(fpath).st_mode | 0o644)
+            except Exception:
+                pass
+
+
+def _tar_extractall_safe(fname: str, open_mode: str, extract_path: str) -> None:
+    try:
+        with contextlib.closing(tarfile.open(fname, open_mode)) as tf:
+            tf.extractall(path=extract_path)
```

Ruff (0.15.15)

[error] Uses of `tarfile.extractall()`

(S202)